protocol/app/upgrades/v.cosmwasm.0/upgrade.go

```diff
@@ -0,0 +1,164 @@
+func InitializeModuleAccs(ctx sdk.Context, ak authkeeper.AccountKeeper) {
+	for _, modAccName := range ModuleAccsToInitialize {
+		// Get module account and relevant permissions from the accountKeeper.
+		addr, perms := ak.GetModuleAddressAndPermissions(modAccName)
+		if addr == nil {
+			panic(fmt.Sprintf(
+				"Did not find %v in `ak.GetModuleAddressAndPermissions`. This is not expected. Skipping.",
+				modAccName,
+			))
+		}
+
+		// Try to get the account in state.
+		acc := ak.GetAccount(ctx, addr)
+		if acc != nil {
+			// Account has been initialized.
+			macc, isModuleAccount := acc.(sdk.ModuleAccountI)
+			if isModuleAccount {
+				// Module account was correctly initialized. Skipping
+				ctx.Logger().Info(fmt.Sprintf(
+					"module account %+v was correctly initialized. No-op",
+					macc,
+				))
+				continue
+			}
+			// Module account has been initialized as a BaseAccount. Change to module account.
+			// Note: We need to get the base account to retrieve its account number, and convert it
+			// in place into a module account.
+			baseAccount, ok := acc.(*authtypes.BaseAccount)
+			if !ok {
+				panic(fmt.Sprintf(
+					"cannot cast %v into a BaseAccount, acc = %+v",
+					modAccName,
+					acc,
+				))
+			}
+			newModuleAccount := authtypes.NewModuleAccount(
+				baseAccount,
+				modAccName,
+				perms...,
+			)
+			ak.SetModuleAccount(ctx, newModuleAccount)
+			ctx.Logger().Info(fmt.Sprintf(
+				"Successfully converted %v to module account in state: %+v",
+				modAccName,
+				newModuleAccount,
+			))
+			continue
+		}
+
+		// Account has not been initialized at all. Initialize it as module.
+		// Implementation taken from
+		// https://github.com/dydxprotocol/cosmos-sdk/blob/bdf96fdd/x/auth/keeper/keeper.go#L213
+		newModuleAccount := authtypes.NewEmptyModuleAccount(modAccName, perms...)
+		maccI := (ak.NewAccount(ctx, newModuleAccount)).(sdk.ModuleAccountI) // this set the account number
+		ak.SetModuleAccount(ctx, maccI)
+		ctx.Logger().Info(fmt.Sprintf(
+			"Successfully initialized module account in state: %+v",
+			newModuleAccount,
+		))
+	}
+}
+
+func removeStatefulFOKOrders(ctx sdk.Context, k clobtypes.ClobKeeper) {
+	allStatefulOrders := k.GetAllStatefulOrders(ctx)
+	for _, order := range allStatefulOrders {
+		if order.TimeInForce == clobtypes.Order_TIME_IN_FORCE_FILL_OR_KILL {
+			// Remove the orders from state.
+			k.MustRemoveStatefulOrder(ctx, order.OrderId)
+
+			// Send indexer event for removal of stateful order.
+			k.GetIndexerEventManager().AddTxnEvent(
+				ctx,
+				indexerevents.SubtypeStatefulOrder,
+				indexerevents.StatefulOrderEventVersion,
+				indexer_manager.GetBytes(
+					indexerevents.NewStatefulOrderRemovalEvent(
+						order.OrderId,
+						indexershared.ConvertOrderRemovalReasonToIndexerOrderRemovalReason(
+							clobtypes.OrderRemoval_REMOVAL_REASON_CONDITIONAL_FOK_COULD_NOT_BE_FULLY_FILLED,
+						),
+					),
+				),
+			)
+		}
+	}
+}
+
+func initRevShareModuleState(
```

Comment:
Why is this needed? This upgrade handler should handle CosmWasm migrations only

Reply:
discussed in https://github.com/dydxprotocol/v4-chain/pull/1708#discussion_r1670834232